File: frontend/lib/models/user_medkit.dart
import 'user.dart';
import 'medkit.dart';

class UserMedKit {
  final int id;
  final int idUser;
  final int idMedKit;
  final User user;
  final MedKit medKit;

  UserMedKit({
    required this.id,
    required this.idUser,
    required this.idMedKit,
    required this.user,
    required this.medKit,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'id_user': idUser,
      'id_med_kit': idMedKit,
      'user': user.toJson(),
      'med_kit': medKit.toJson(),
    };
  }

  factory UserMedKit.fromJson(Map<String, dynamic> json) {
    return UserMedKit(
      id: json['id'],
      idUser: json['id_user'],
      idMedKit: json['id_med_kit'],
      user: User.fromJson(json['user']),
      medKit: MedKit.fromJson(json['med_kit']),
    );
  }

  @override
  String toString() {
    return 'UserMedicineKit{id: $id, idUser: $idUser, idMedKit: $idMedKit}';
  }
}

File: backend/models/user_med_kit.py
from sqlalchemy import Column, Integer, Sequence, ForeignKey
from sqlalchemy.orm import relationship
from backend.database import Base


class UserMedicineKit(Base):
    __tablename__ = "user_med_kit"

    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    id_user = Column(
        Integer, ForeignKey("user.id_user", ondelete="CASCADE"), nullable=False
    )
    id_med_kit = Column(
        Integer, ForeignKey("med_kit.id_med_kit", ondelete="CASCADE"), nullable=False
    )

    user = relationship("User", backref="user_med_kits")
    med_kit = relationship("MedicineKit", backref="med_kit_users")

    def __repr__(self):
        return f"<UserMedicineKit(id_user={self.id_user}, id_med_kit={self.id_med_kit})>"
